tests/validation/tests/v3_api/common.py

```diff
@@ -2084,15 +2084,28 @@ def readDataFile(data_dir, name):
         return f.read()
 
 
-def set_url_password_token(rancher_url, server_url=None):
+def set_url_password_token(rancher_url, server_url=None, version="oldUI"):
     """Returns a ManagementContext for the default global admin user."""
     auth_url = \
         rancher_url + "/v3-public/localproviders/local?action=login"
-    r = requests.post(auth_url, json={
-        'username': 'admin',
-        'password': 'admin',
-        'responseType': 'json',
-    }, verify=False)
+    rpassword = 'admin'
+    print(auth_url)
+    if version.find("master") > -1 or version.find("2.6") > -1:
+        rpassword = ADMIN_PASSWORD
+        print("on 2.6 or later")
+    retries = 4
+    for attempt in range(retries):
+        try:
+            r = requests.post(auth_url, json={
+            'username': 'admin',
+            'password': rpassword,
+            'responseType': 'json',
+            }, verify=False)
+            break
+        except:
```

Comment:
A naked `except` is a bad pattern, this should name the error expected so if it isn't of that type the exception actually happens. i.e. `except ApiError:`

Reply:
I've updated this to `except OSError` as this was the original error being thrown. rerun of the jenkins job with the updated code is passing

Reply:
`OSError` is not quite specific for the `requests` library `ConnectionError` would be more appropriate.